Release notes and additional documentation for Forms 13.3 and 14.2. (#6532)
umbraco/UmbracoDocs: 13/umbraco-forms/developer/themes.md

From Forms 13.3 it's possible to do this for custom themes.

Firstly you'll create a new Razor Class Library project to hold the theme.  You then create the necessary partial views for your theme as usual within `Views\Partials\Forms\Themes\<my-custom-theme>`.

You then need to provide the names of the files in your theme via an implementation of `ITheme`. For example, if only overriding a single file, your class would look like this:

```csharp
using Umbraco.Forms.Core.Interfaces;

public class MyCustomTheme : ITheme
{
    private const string FilePathFormat = "{0}/{1}/{2}.cshtml";

    public virtual string Name => "my-custom-theme";

    public virtual IEnumerable<string> Files =>
        [
            string.Format(FilePathFormat, Core.Constants.System.ThemesPath, Name, "FieldTypes/FieldType.Textfield"),
        ];
}
```

In your project that consumes the theme, you register the ones you want to use via a composer:

```csharp
public class MyComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
      builder.Themes()
          .Add<MyCustomTheme>();
    }
}
```

> With that in place your theme will be picked up for selection in the theme picker. And the partial view files included will be used when rendering forms.

Reviewer: ```suggestion
Your theme will now be available in the Theme Picker and the partial view files will be used when rendering forms.
```

#### Email Templates

> Email templates provided for the send email workflow can be provided in a Razor Class Library in a similar way.

Reviewer: ```suggestion
Email templates provided for the send email workflow can be provided in a Razor Class Library similar to the Theme files.
```
